tensorflow_addons/optimizers/tests/weight_decay_optimizers_test.py

```diff
@@ -212,129 +173,159 @@ def sgdw_update_numpy(param, grad_t, slot_vars, learning_rate, momentum, weight_
 
-    def testBasicCallableParams(self):
-        self.doTest(
-            self.optimizer,
-            sgdw_update_numpy,
-            learning_rate=lambda: 0.001,
-            momentum=lambda: 0.9,
-            weight_decay=lambda: WEIGHT_DECAY,
-        )
+@pytest.mark.usefixtures("maybe_run_functions_eagerly")
+@pytest.mark.parametrize("dtype", [(tf.half, 0), (tf.float32, 1), (tf.float64, 2)])
+def test_basic_decay_var_list_adamw(dtype):
+    do_test(
+        dtype,
+        optimizer_AdamW,
+        adamw_update_numpy,
+        do_decay_var_list=True,
+        learning_rate=0.001,
+        beta_1=0.9,
+        beta_2=0.999,
+        epsilon=1e-8,
+        weight_decay=WEIGHT_DECAY,
+    )
+
+
+def test_keras_fit():
+    """Check if calling model.fit works."""
+    model = tf.keras.models.Sequential([tf.keras.layers.Dense(2)])
+    loss = tf.keras.losses.SparseCategoricalCrossentropy(from_logits=True)
+    optimizer = optimizer_AdamW(learning_rate=1e-4, weight_decay=1e-4)
+    model.compile(optimizer=optimizer, loss=loss, metrics=["accuracy"])
+    X, y = np.random.uniform(size=(2, 4, 1))
+    model.fit(X, y, epochs=1)
 
-    def testBasicDecayVarList(self):
-        self.doTest(
-            self.optimizer,
-            sgdw_update_numpy,
-            do_decay_var_list=True,
-            learning_rate=0.001,
-            momentum=0.9,
-            weight_decay=WEIGHT_DECAY,
-        )
 
+optimizer_SGDW = weight_decay_optimizers.SGDW
+
+
+@pytest.mark.usefixtures("maybe_run_functions_eagerly")
+@pytest.mark.parametrize("dtype", [(tf.half, 0), (tf.float32, 1), (tf.float64, 2)])
+def test_sparse_sgdw(dtype):
+    do_test(
+        dtype,
+        optimizer_SGDW,
+        sgdw_update_numpy,
+        do_sparse=True,
+        learning_rate=0.001,
+        momentum=0.9,
+        weight_decay=WEIGHT_DECAY,
+    )
+
+
+@pytest.mark.usefixtures("maybe_run_functions_eagerly")
+@pytest.mark.parametrize("dtype", [tf.half, tf.float32, tf.float64])
+def test_sparse_repeated_indices_sgdw(dtype):
+    do_test_sparse_repeated_indices(
+        dtype,
+        optimizer_SGDW,
+        learning_rate=0.001,
+        momentum=0.9,
+        weight_decay=WEIGHT_DECAY,
+    )
+
+
+@pytest.mark.usefixtures("maybe_run_functions_eagerly")
+@pytest.mark.parametrize("dtype", [(tf.half, 0), (tf.float32, 1), (tf.float64, 2)])
+def test_basic_sgdw(dtype):
+    do_test(
+        dtype,
+        optimizer_SGDW,
+        sgdw_update_numpy,
+        learning_rate=0.001,
+        momentum=0.9,
+        weight_decay=WEIGHT_DECAY,
+    )
+
+
+@pytest.mark.usefixtures("maybe_run_functions_eagerly")
+@pytest.mark.parametrize("dtype", [(tf.half, 0), (tf.float32, 1), (tf.float64, 2)])
+def test_basic_callable_params_sgdw(dtype):
+    do_test(
+        dtype,
+        optimizer_SGDW,
+        sgdw_update_numpy,
+        learning_rate=lambda: 0.001,
+        momentum=lambda: 0.9,
+        weight_decay=lambda: WEIGHT_DECAY,
+    )
 
-class ExtendWithWeightDecayTest(SGDWTest):
```

Comment:
The equivalent of this would be to run the SGDWTest tests with the decorator:

```python
@pytest.mark.parametrize(
    "optimizer", 
    [weight_decay_optimizers.SGDW, 
     weight_decay_optimizers.extend_with_decoupled_weight_decay(tf.keras.optimizers.SGD)])
```

Reply:
Thanks @gabrieldemarmiesse. Updated, needs your review again.